src/game/Pet.cpp

```diff
@@ -916,6 +916,8 @@ void Pet::Unsummon(PetSaveMode mode, Unit* owner /*= nullptr*/)
 
 void Pet::GivePetXP(uint32 xp)
 {
+    xp *= sWorld.getConfig(CONFIG_FLOAT_RATE_XP_PETKILL);
```

Comment:
This should be conditioned and if no rate is found by getConfig() then fall back to a default rate of 1, like we do with most other configurable rates.